src/Microsoft.Data/BaseColumn.cs

```diff
@@ -41,7 +41,17 @@ protected set
             get;
         }
 
-        public string Name { get; set; }
+        private string _name;
+        public string Name => _name;
+
+        public void SetName(string newName, DataFrame dataFrame = null)
+        {
+            if (!ReferenceEquals(dataFrame, null))
```

Comment:
Why are you sometimes using `x is null` and sometimes using `ReferenceEquals(x, null)`. We should be consistent in how we are checking for null.

Reply:
That's because I overloaded the `==` operator on DataFrame to mean element wise equality. I've since thought many times that writing `ReferenceEquals` to check for null doesn't feel right, but `==` does exist in Pandas

Reply:
You can use ‘x is null’ in all cases and it will skip the == operator.